### PR TITLE
Replace patch with upstream PR

### PR DESCRIPTION
**Pull request by chrisburr**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Use the patches from https://github.com/root-project/root/pull/4010 and https://github.com/root-project/root/pull/4011

### CODE REVIEW

**Comment by henryiii** [comment]
```
x86_64-conda_cos6-linux-gnu-c++: error: unrecognized command line option '-std=c++-std=c++1z'; did you mean '--std=c++1z'?
```

File: recipe/patches/root-6.18.02-dont-use-cmake-cxx-standard.patch [hunk 1]
```diff
@@ -1,25 +1,22 @@
...
+index dcf3c5aea07..2eb27b5611d 100644
 --- a/cmake/modules/CheckCompiler.cmake
 +++ b/cmake/modules/CheckCompiler.cmake
 @@ -115,7 +115,7 @@ if(NOT CMAKE_CXX_STANDARD MATCHES "11|14|17")
  endif()
  
  # needed by roottest, to be removed once roottest is fixed
 -set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++${CMAKE_CXX_STANDARD}")
-+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++1z")
++set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++${CMAKE_CXX${CMAKE_CXX_STANDARD}_STANDARD_COMPILE_OPTION}")
```
Comment (henryiii) [inline recipe/patches/root-6.18.02-dont-use-cmake-cxx-standard.patch]:
```suggestion
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_CXX${CMAKE_CXX_STANDARD}_STANDARD_COMPILE_OPTION}")
```

Probably in the ROOT patch too?

Comment (chrisburr) [reply recipe/patches/root-6.18.02-dont-use-cmake-cxx-standard.patch]: Yes I did 👍

Comment (henryiii) [reply recipe/patches/root-6.18.02-dont-use-cmake-cxx-standard.patch]: Will this be in a PR to ROOT?

**Comment by henryiii** [comment]
Clear to merge?

**Comment by chrisburr** [comment]
I'm trying to see if I can get the split package working (it's almost there but Linux fails for some reason, see #34).

**Comment by chrisburr** [comment]
I've added a hack which should fix #41, if this passes and the log contains:
```
Found a 64bit system
```
instead of what is currently shown for 6.18:
```
Found a 32bit system
```
it should be merged.

**Pull request merged**